docker/configure_workers_and_start.py

```diff
@@ -271,7 +319,7 @@ def convert(src: str, dst: str, **template_vars: object) -> None:
         outfile.write(rendered)
 
 
-def add_sharding_to_shared_config(
+def process_sharding_and_stream_writers_for_shared_config(
```

Comment:
since you wanted a name idea, `add_worker_roles_to_shared_config` would encompass both sharding and stream writer and potentially other kinds of roles as well

Reply:
I like your suggestion. Going with that.